doc/stdenv/stdenv.xml

```diff
@@ -2111,16 +2111,105 @@ postInstall = ''
       </para>
      </listitem>
     </varlistentry>
+
     <varlistentry>
-     <term>
-      cmake
+     <term xml:id="cmake">
+       cmake
      </term>
      <listitem>
       <para>
-       Overrides the default configure phase to run the CMake command. By default, we use the Make generator of CMake. In addition, dependencies are added automatically to CMAKE_PREFIX_PATH so that packages are correctly detected by CMake. Some additional flags are passed in to give similar behavior to configure-based packages. You can disable this hook’s behavior by setting configurePhase to a custom value, or by setting dontUseCmakeConfigure. cmakeFlags controls flags passed only to CMake. By default, parallel building is enabled as CMake supports parallel building almost everywhere. When Ninja is also in use, CMake will detect that and use the ninja generator.
+       Overrides the default configure phase to run the CMake command. By default, we use the Make generator of CMake, however, if ninja is present, then the ninja generator will be used instead. In addition, dependencies are added automatically to CMAKE_PREFIX_PATH so that packages are correctly detected by CMake. By default, the cmake configurePhase can be thought of as:
```

Comment:
I think it's still worthwhile mentioning somewhere in the section that `enableParallelBuilding = true` by default for CMake, since there are still regularly PRs with `enableParallelBuilding = true` when the derivation uses CMake.